core/transaction.c

```diff
@@ -234,11 +234,52 @@ void transaction_handle_response(lwm2m_context_t * contextP,
                 // So we resend transaction that were denied for authentication reason.
                 if (message->code != COAP_401_UNAUTHORIZED || transacP->retrans_counter >= COAP_MAX_RETRANSMIT)
```

Comment:
For the record, this HACK will be removed for a better one:
When a client receives a CON packet from a server in REG_PENDING_STATE, it will silently ignore it. Thus the server will retransmit the CON and it should arrive after the registration ACK.